### PR TITLE
Hotfix/jsdocs gen invite

### PR DESCRIPTION
**Pull request by NeerjaGoyalNHS**
## Description
JS docs for generateInvitesTriggerLambda

## Type of changes

- [x] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 1]
```diff
@@ -108,7 +117,17 @@ export const handler = async (event) => {
   return "Exiting Lambda";
 };
 
-// METHODS
+/**
+ * Reads a CSV file from S3.
+ * @async
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:122]: Leave a space between definition and other declarations

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 2]
```diff
@@ -248,7 +321,14 @@ export const checkDynamoTable = async (
   }
 };
 
-// returns formatted dynamodb json to be uploaded or rejected object
+/**
+ * @async
+ * @function generateRecord
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:326]: no definition?

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 3]
```diff
@@ -280,8 +360,16 @@ const generateRecord = async (record, client) => {
   return await formatDynamoDbRecord(record);
 };
 
-// returns unique participant id using NHS regex
-// unique in both population table and this stream
+/**
+ * Generates unique participant id using NHS regex
+ * unique in both population table and this stream
+ * @async
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:366]: space missing

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 4]
```diff
@@ -399,7 +506,18 @@ export const getItemFromTable = async (dbClient, table, ...keys) => {
   return response;
 };
 
-// returns successful response if attribute doesn't exist in table
+/**
+ * This function allows the user to query against DynamoDB.
+ * @async
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:511]: space here as well please

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 5]
```diff
@@ -436,7 +554,19 @@ export const lookUp = async (dbClient, ...params) => {
   return SUCCESSFULL_REPSONSE;
 };
 
-// returns response from batch write to dynamodb table
+//
```
Comment (Abdul-NHS) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:557]: remove

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:1]: caasFeedAddRecordsLambda is also in PR 662 and review comments have been added in that PR. Please address comments for this lambda in PR 662 and merge into this PR.

File: terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js [hunk 6]
```diff
@@ -11,6 +11,16 @@ const client = new DynamoDBClient({ region: "eu-west-2" });
 const ENVIRONMENT = process.env.ENVIRONMENT;
 const SUCCESSFULL_REPSONSE = 200;
 
+/**
+ * AWS Lambda handler to load participant info for participants with result - CSD.
```
Comment (chenglawson1) [inline terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js:15]: Lambda is to set invited participant status, not to load participant info with CSD result.

File: terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js [hunk 7]
```diff
@@ -11,6 +11,16 @@ const client = new DynamoDBClient({ region: "eu-west-2" });
...
+ *
+ * @async
+ * @function handler
+ * @param {Object} event - The event triggering the Lambda.
+ * @param {Object} context - The context object provided by AWS Lambda.
+ * @returns {Promise<Object>} - A promise that resolves to the response object containing participant info or an error message.
```
Comment (chenglawson1) [inline terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js:21]: Response object contains success message or error details.

File: terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js [hunk 8]
```diff
@@ -83,9 +93,17 @@ export const handler = async (event, context) => {
...
-// assign it to records array
+/**
+ * create a batch id and assign it to records array
+ *
+ * @async
+ * @function updatePersonsToBeInvited
+ * @param {Array<Object>} recordArray - Array of participants records.
+ * @param {String} createdBy createdBy info
+ * @param {DynamoDBClient} client Instance of DynamoDB client
+ * @returns {number} The HTTP status code of the update operation.
```
Comment (chenglawson1) [inline terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js:104]: Return type is not the http status code number, it is a Promise of array of each update result.

File: terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js [hunk 9]
```diff
@@ -83,9 +93,17 @@ export const handler = async (event, context) => {
   }
 };
 
-// METHODS
-// create a batch id
-// assign it to records array
+/**
+ * create a batch id and assign it to records array
```
Comment (chenglawson1) [inline terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js:97]: Update participants status to "to be invited" and assign a batch id in Population table.

File: terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js [hunk 10]
```diff
@@ -142,6 +172,16 @@ export async function updateRecord(record, batchId, client, createdBy) {
...
 }
 
+/**
+ * gets LSOA from the personID from Poulation Table
+ *
+ * @async
+ * @function getLsoaCode
+ * @param {String} record - personID of participant.
+ * @param {DynamoDBClient} client Instance of DynamoDB client
+ * @returns {Object} The HTTP response object.
```
Comment (chenglawson1) [inline terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js:182]: Query command response object, not HTTP.

File: terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js [hunk 11]
```diff
@@ -160,6 +200,21 @@ export async function getLsoaCode(record, client) {
   return response;
 }
 
+/**
+ * Takes single record from Population DynamoDB table and
+ * update that individual to have a identifiedToBeInvited field set to true
```
Comment (chenglawson1) [inline terraform/src/generateInvitesTriggerLambda/generateInvitesTriggerLambda.js:205]: Description is copy and paste error of above function.

File: terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js [hunk 12]
```diff
@@ -26,7 +27,15 @@ const UNSUCCESSFULL_REPSONSE = 400;
...
 let participantIdStore = {};
 
-// Lambda Entry Point
+/**
+ * AWS Lambda handler to load eligible participant info to DynamoDB.
+ *
+ * @async
+ * @function handler
+ * @param {Object} event - The event triggering the Lambda.
+ * @returns {Object} - A promise that resolves to the response object containing participant info or an error message.
```
Comment (chenglawson1) [inline terraform/src/caasFeedAddRecordsLambda/caasFeedAddRecordsLambda.js:36]: Comment in PR 662 not resolved.

**Review by chenglawson1 — APPROVED** [review]

**Review by Aymene-n — APPROVED** [review]

**Pull request merged**